src/main/java/org/opensearch/knn/index/codec/KNN910Codec/KNN910Codec.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.knn.index.codec.KNN910Codec;
+
+import org.apache.lucene.codecs.Codec;
+import org.apache.lucene.codecs.CompoundFormat;
+import org.apache.lucene.codecs.DocValuesFormat;
+import org.apache.lucene.codecs.FilterCodec;
+import org.opensearch.knn.index.codec.KNNFormatFacade;
+import org.opensearch.knn.index.codec.KNNFormatFactory;
+
+import static org.opensearch.knn.index.codec.KNNCodecFactory.CodecDelegateFactory.createKNN91DefaultDelegate;
+
+/**
+ * Extends the Codec to support a new file format for KNN index
+ * based on the mappings.
+ *
+ */
+public final class KNN910Codec extends FilterCodec {
+
+    private static final String KNN910 = "KNN910Codec";
+    private final KNNFormatFacade knnFormatFacade;
+
+    /**
+     * No arg constructor that uses Lucene91 as the delegate
+     */
+    public KNN910Codec() {
+        this(createKNN91DefaultDelegate());
```

Comment:
Why do this and line 40 need to include version in the name? Couldnt version be passed as an arg to createDefaultDelegate and createKNNFormat?